Create receive page in payments section (#240)
BitcoinDesign/Guide: guide/payments/receive.md

---
layout: guide
title: Receiving bitcoin
description: An introduction to how receiving bitcoin works.
nav_order: 3
parent: Payments
permalink: /guide/payments/receive/
main_classes: -no-top-padding
# image: /assets/images/guide/payments/receive/header.svg
---

# Receiving bitcoin

<!--

Editor's notes

This page should cover what to do when receiving bitcoin, how to share and copy addresses etc.

- What addresses to support (native segwit by default)
- How to present QR codes (allow for specifying an amount)
- Generating a new address 

-->

> The details required by a sender to make a payment, at minimum, is an address. You can think of an address just as you would an invoice, as it is only used once. The receiver generates an address in their Bitcoin application and gives it to the sender. Just as with traditional invoices, a new address needs to be created each time the receiver wants to receive funds.

Reviewer: Great explanation.

> The details required by a sender to make a payment, at minimum, is an address. You can think of an address just as you would an invoice, as it is only used once. The receiver generates an address in their Bitcoin application and gives it to the sender. Just as with traditional invoices, a new address needs to be created each time the receiver wants to receive funds.

Reviewer 2: This is not accurate, as addresses can be used multiple times. I'd differentiate between "can" and "should".

Reviewer: Privacy by default. But I agree perhaps `should` is a better choice of words, still, we need to have strong opinions and recommendations when it comes to these things.

Author: fair points @GBKS @pavlenex , thoughts on this suggestion by @danielnordh?

```suggestion
For someone to receive bitcoin, the sender needs to know the recipient’s address. The receiver generates an address in their Bitcoin application and shares it with the sender, either as plain text, a QR code, or a payment link.  You can think of an address just as you would an invoice, you generate a new one for each transaction, as addresses should only be used once to increase privacy. 
```

Author: minimal change. 

```suggestion
The details required by a sender to make a payment, at minimum, is an address. You can think of an address just as you would an invoice, as it should only be used once. The receiver generates an address in their Bitcoin application and gives it to the sender. Just as with traditional invoices, a new address needs to be created each time the receiver wants to receive funds.
```

Reviewer 2: I like Daniel's as it is written from the perspective of a receiver, and this is the receive page.

Author: True. Good point.

There is other contextual information about payments that can be useful to keep or to communicate, like the names of the transacting parties or the purpose of the transaction. As the blockchain only stores the amount and addresses involved in a transaction, any additional information has to be stored in your application, or encoded in payment links when they need to be shared.

Let's look at our options when receivers want to share other details with the sender, or store them with the payment request for their own record keeping.

<!-- 
Update glossary

Each address that a Bitcoin application generates has an accompaniying private key. This private key allows the receiver to unlock the bitcoin that is sent to the address when the receiver chooses to spend it.

Although addresses are not a problem to share publically, their accompanying private keys which typically generated from a single recovery phrase must be kept secure and private.
-->

> ## Creating an invoice

Author: @GBKS @pavlenex thoughts on referring to generating an address and adding metadata to it as an invoice? I previously called it payment requests, and @danielnordh prefers that as it is less corporate/businessy.

Reviewer 2: I find invoice to be more of business term than payment request (thinking about which terms I'd use with friends vs. business purposes). So my vote would be for payment request (despite it being longer).

Reviewer: I prefer an invoice, easier to translate, the payment request is not a regular term in my opinion.

Author: When we start talking about LN, it's all about invoices there, and some of the additional data is going to be more built-in.

Invoice
- https://en.bitcoin.it/wiki/Invoice_address
- https://github.com/lightningnetwork/lightning-rfc/blob/master/11-payment-encoding.md

#### Payment Request
- https://en.bitcoin.it/wiki/Payment_Request

👆 this is old deprecated BIP but still has some good ideas around signed messaged that [will probably be rehashed](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-March/018560.html).

Reviewer 3: I commented a bit more on this part in the [doc](https://docs.google.com/document/d/1PMgMIZ1fkC3SqJKWQBOrmJxefjmZhktqbSPZfzabmrE/edit?usp=sharing)

My feedback is basically:

- If 'invoice' is not a defined standard in Bitcoin, let's not push for it as it doesn't really make sense for end users (maybe for merchants, but that's not what we are tackling here)
- We don't not need to talk about this as creating an 'object', so instead of saying 'Create an invoice/payment request', we can simply say 'Requesting bitcoin' as a subtitle. For the user it's an action, not an object.
- IF the user is asked/able to enter additional information other than the address, it should be clear to them if that information is SHARED with the person they want to request bitcoin from, or not

Reviewer 2: Good point about this object-creation idea. I prefer the simpler language that avoids that concept and focuses on the action.

Author: I wouldn't dismiss the object creation idea - as sending bitcoin is very much action oriented flow, the act of requesting bitcoin has artefacts that the designer would need to consider in other areas of the application and would benefit from understanding that object and its properties. So I think in this case it would be very well suited.

Since this comment though, the state of the page has gone towards something which is more action oriented, I'll probably bring it up at another point though after the chapter complete for v1.

Oftentimes, to complete a payment, the sender may need to know the specific amount to be sent. On the receiver’s side, they may want to keep some notes about who sent the bitcoin to help them manage their funds in the future and understand their past transaction activity.

The following are the primary properties of a Bitcoin invoice:

- **Address** – The destination of the payment
- **Amount** – How much is being requested (optional)
- **From** – Name of who is sending the payment (optional)
- **Purpose/Message** – Note with a description of what the payment is for (optional)

While the application will generate a new address in the background, other information like; how much is being requested, the name of the sender, and what is being paid for has to be manually entered by the receiver. The order in which these properties get entered is up to you, and besides the address, all else is optional.

<!--
Technical nitpick;
Addresses are not exactly "generated"
since they are deterministic technically
what happens is an unused address just gets chosen
-->

> The more time passes, the more difficult it will be to recall the details of transactions. When transacting in Bitcoin there is no requirement for senders and receivers even to identify themselves, so both sides may end up having a transaction history with only amounts and random-looking text (addresses) to reference when trying to understand where some funds came from.

Reviewer: ```suggestion
The more time passes, the more difficult it will be to recall the details of transactions. When transacting in Bitcoin, there is no requirement for senders and receivers even to identify themselves, so both sides may end up having a transaction history with only amounts and random-looking text (addresses) to reference when trying to understand where some funds came from.
```

   retina = "/assets/images/guide/payments/receive/enter-details-from@2x.png"
   alt-text = "Screen asking the receiver to enter the name of the sender"
   caption = "The receiver can specify the amount that they are requesting. This could be shared in payment links which would auto-fill the amount for the sender."
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/payments/receive/enter-details-purpose.png"
   retina = "/assets/images/guide/payments/receive/enter-details-purpose@2x.png"
   alt-text = "Screen asking the receiver to enter the purpose of the payment"
   caption = "Sometimes the purpose can be determined by identifying who the payment is from. Be mindful of that, and consider easy ways to help the receiver add more context."
   width = 250
   height = 541
%}
</div>

If your design’s goal is to get the receiver in the quickest time to share an address, you can choose to show them a new address straight away. The other properties can be left as optional fields. Consider the trade-off when optimizing this way, as there will be less useful contextual details for reports that you may want to provide in the future.

> <div class="image-slide-gallery">

Reviewer 2: No need for the slide gallery if you'll have only one image. You can just use the picture include with "layout=background". Could also go for the split-column layout.

Author: Nice, the -background class needs to have a text-align: center though, shall I include that in this PR?

<img width="677" alt="Screenshot 2021-04-29 at 13 53 58" src="https://user-images.githubusercontent.com/183140/116546804-5b2a8f80-a8f2-11eb-8d72-4dcbf0f214ab.png">

<img width="812" alt="Screenshot 2021-04-29 at 13 53 07" src="https://user-images.githubusercontent.com/183140/116546719-3d5d2a80-a8f2-11eb-8d0a-eeb43feb7fa4.png">

Author: hmm - i think i mistook picture.html for image.html

Reviewer 2: No. Conor is already adding this in #225. That's the reason why we need to separate content commits and code commits.

Author: been having issues with this picture/image component, there are no captions. im leaving it as a gallery for now - unless you can walk me through how to get it working as you have in mind.

{% include picture.html
   image = "/assets/images/guide/payments/receive/enter-details-compact.png"
   retina = "/assets/images/guide/payments/receive/enter-details-compact@2x.png"
   alt-text = "Screen showing QR Code for the address and optional entry for other properties"
   caption = "All properties editable on a single screen"
   width = 250
   height = 541
%}

</div>

<!-- 
Update interopability page
Since there are many Bitcoin applications for people to choose from, its likely that the sender and receiver are not using the same one and they don't have the abilitiy to exchange payment details within the application itself. 

The exchange of payment details then most commonly happens outside of the wallet application. This 

Since bitcoin is a open system and has many payment applications built ontop of it, there is a [standard format](https://bitcoin.design/guide/foundations/wallet-interoperability/#payment-links) of "payment links" that most bitcoin applications use to share payment details.
-->

## Choosing the type of address

> There are currently [three different versions](https://bitcoin.design/guide/glossary/#address) of bitcoin addresses. In order of oldest to newest they are; Legacy, Script, and SegWit. It is best practice for new applications to support SegWit, the latest, by default.

Reviewer: Is there a reason why you decided to call this `Script`? It sounds generic. SegWit-wrapped? Curious on the source why it's `script`, is it because of `PWSH`?

Reviewer: A good way to split these visually is:
- Legacy starts with 1
- P2SH starts with 3
- SegWit starts with BC1

Author: @pavlenex its from the glossary actually. i didn't know it was called script before, since most address types going forward are script based.

Author: moved this to the glossary under "address compatibility" subheading of the address definition

> There are currently [three different versions](https://bitcoin.design/guide/glossary/#address) of bitcoin addresses. In order of oldest to newest they are; Legacy, Script, and SegWit. It is best practice for new applications to support SegWit, the latest, by default.

Reviewer 2: Change to relative link.

> The most commonly used addresses are still the “legacy” addresses since Segwit addresses are only a few years old, and some older applications have yet to update.

Reviewer: Source?

Reviewer: I feel like we should recommend a particular address type here - SegWit with a fallback or switch option, I don't think we have enough information here nor that users will be able to decide for themselves what to use as we explained possible causes only on the surface (which is enough for the design guide imo), just we should maybe do it in `DO` recommend BC1 with the option to switch.

Author: Perhaps stated too strongly. Still, non SegWit transactions still make up a large enough amount of transactions that it may become an occasional problem.

- https://blockchair.com/bitcoin/charts/segwit-usage

Author: I saw some places the "Script" address is referred to as "Compatibility" or "SegWit Compatability" address. Perhaps we should consider this. I'd roll that into a bigger update of the glossary address section though.

Suppose the sender is using a legacy wallet and paying to the receiver's SegWit address. In this case, the sender's wallet may incorrectly warn them that the address is invalid or not supported. This can confuse the sender’s end, leading them to think that the receiver provided an incorrect address.

> The receiver should then have the ability to switch to a Script address that does not have all the benefits of SegWit, like cheaper transactions but will be compatible with the sender's wallet.

Reviewer: We should clearly give directions here, I don't think it's clear people on what to use.

Reviewer 2: This is already in one of the screens. I created another one that alerts a user that the format of the address they entered is not supported. [Check it](https://www.figma.com/file/VB3GQdAnhl8yta44DY3PSV/Bitcoin-Wallet-UI-Kit?node-id=1366%3A32884).

Author: @GBKS I think that graphic would more be suited on the send page. Which can be updated to reflect this same issue from the sender's perspective.

Reviewer 2: Sounds good. Feel free to use (and not use) those graphics as you think works best.

Author: thx @GBKS added it

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/payments/receive/share.png"
   retina = "/assets/images/guide/payments/receive/share@2x.png"
   alt-text = "Screen showing the invoice with a share button"
   caption = ""
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/payments/receive/share-sharesheet.png"
   retina = "/assets/images/guide/payments/receive/share-sharesheet@2x.png"
   alt-text = "Screen showing the invoice overlayed with a share sheet"
   caption = ""
   width = 250
   height = 541
%}

</div>

**Text**

Sharing payment details as text is as straightforward as copying and pasting. It is also typical to provide the receiver with a button to conveniently share in other applications, with “share sheets,” which most mobile operating systems offer.

**QR Codes**

QR Codes are also a common way to exchange payment details when the two devices are near one another. It is not without its faults though, as some people may not be familiar with QR Codes, and lower-end phones may have difficulty reading them.

> Here are some constraints to consider when displaying QR Codes:

Reviewer: Might be something worth linking to https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses

Author: added

- QR Codes with more blocks (modules) are challenging to read for devices with low-resolution cameras.
- There should be sufficient contrast between the blocks in a QR Code and the background. Ideally, keeping the blocks black and background white would be the most optimal.
- Size matters. You won't know the distance the devices will be from one another when the scan is happening, so you should design assuming the longest scan distance.

> ## Waiting and transaction processing

Reviewer: We probably need an illustration here.

Reviewer 2: Even though the screens on the page already show QR codes?

Once the payment details are shared, until the sender creates a transaction and broadcast it, there is uncertainty of when the payment would be completed. Think about how to keep the receiver informed while they are waiting.

Similar to [the processes a sender](https://bitcoin.design/guide/payments/send/#transaction-processing-and-confirmation) sees, with the inclusion of this waiting period, here is what the receiver can expect:

**Awaiting payment** -- the sender has not broadcasted a transaction with the payment, or one has not been detected as yet by the receiver's application.

**Pending/Unconfirmed** -- miners know about the transaction, but have yet to process it. 

**First confirmation** -- at this point the receiver may already be able to spend the funds, but has to be mindful of [chain reorganizations](https://bitcoin.design/guide/payments/transactions/#7-confirmations).

**Sixth confirmation** -- from this point the payment is finalized, it should be no problem to spend freely.

> If your application has an area where the user can see a list of payment requests, this would be a good place to use it to also indicate which specific stage the payment is in.

Reviewer 2: There are several screens with lists of transaction in the wallet UI kit we can use here. [Check it](https://www.figma.com/file/VB3GQdAnhl8yta44DY3PSV/Bitcoin-Wallet-UI-Kit?node-id=1339%3A32742).

Although addresses are not a problem to share publically, their accompanying private keys which typically generated from a single recovery phrase must be kept secure and private.
-->

## Preparing the payment request
The simplest way to request bitcoin is to create and share a Bitcoin address where the sender should direct the payment. Generating a new address and showing it as a QR code and plain text is the most common way for applications to help users do so.

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/payments/receive/enter-details-compact.png"
   retina = "/assets/images/guide/payments/receive/enter-details-compact@2x.png"
   alt-text = "Screen showing QR Code for the address and optional entry for other properties"
   caption = "The receiver is able to share a bitcoin address as soon as they enter the request flow"
   width = 250
   height = 541
%}

</div>

{% include tip/tip.html %}
Since there are multiple types of addresses, there are some compatibility issues between them that can disrupt a payment flow. Read more about [address compatibility](#address-compatibility).
{% include tip/close.html %}

> The receiver may want to request a specific amount from the sender, or save additional information for their own record. This will require extra steps and information when generating the payment request, which can then be shared with the sender as a payment link.

Reviewer 4: When you say link are you referring to URI? https://en.bitcoin.it/wiki/BIP_0021. If so these can be shared with the sender as a payment link (URI) or QR.

-->

## Sharing the payment request
Whether it’s just an address or a payment link, the two methods of sharing the payment details are text or scannable QR Codes.

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/payments/receive/share.png"
   retina = "/assets/images/guide/payments/receive/share@2x.png"
   alt-text = "Screen showing the invoice with a share button"
   caption = ""
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/payments/receive/share-sharesheet.png"
   retina = "/assets/images/guide/payments/receive/share-sharesheet@2x.png"
   alt-text = "Screen showing the invoice overlayed with a share sheet"
   caption = ""
   width = 250
   height = 541
%}

</div>

**Text**

Sharing payment details as text is as straightforward as copying and pasting. It is also typical to provide the receiver with a button to conveniently share in other applications, with “share sheets,” which most mobile operating systems offer.

**QR Codes**

> QR Codes are also a common way to exchange payment details when the two devices are near one another. It is not without its faults though, as some people may not be familiar with QR Codes, and lower-end phones may have difficulty reading them.

Reviewer 4: QR codes aren't just used when devices are close together.  They can be shared only and sometimes sent as an image.

Author: hmm what do you mean by shared only, like printed or something? Saw a video of some [drones making a QR](https://www.youtube.com/watch?v=G9K9OOAcUWQ) in the sky. In this case it isn't a device with a wallet presenting the QR.

As an image though, for it to be scanned by the camera it still needs to be close to the other device that is presenting the QR. If the QR is on the same device that has to read it, then another form of input data would be more suitable, especially in mobile contexts (text).

Author: Help me understand the argument a bit more though, cause not sure I've gotten it, and I'd like to get the QR stuff correct since its so critical in bitcoin world.

Reviewer 4: Meant to say shared online* sorry, which would encompass being sent as an image. Here is an example of this being used:

Receiver generates new address / associated QR. Receiver and sender are not close by so the sender scanning the receivers QR off their device isn't possible. Receiver emails sender the address / QR. Sender scans the QR they received via email (most likely opening the email on their desktop and scanning with their mobile wallet.)

Some wallets, Bluewallet being an example, can actually send to an address through a QR image saved locally in your photos. So the receiver would send the QR as an image, you click send in blue wallet, then click the 'image' icon in bottom left, and click on the image of the QR. On mobile bitcoin payment URI that openly directly in your wallet are probably a better UX but this alternative option still exists.

_For more technical details about optimizing   QR Codes [see this article](https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses)._

## Awaiting the payment
Once the payment details are shared, until the sender creates a transaction and broadcast it, there is uncertainty of when the payment would be completed. Think about how to keep the receiver informed while they are waiting.

Similar to [the processes a sender]({{ '/guide/payments/send/#transaction-processing-and-confirmation' | relative_url }}) sees, with the inclusion of this waiting period, here is what the receiver can expect:

[WHAT DOES THE RECEIVER SEE?]

If your application has an area where the user can see a list of payment requests, this would be a good place to use it to also indicate which specific stage the payment is in.

Then once the payment has been finalized, you should also consider what the receiver may want to do with those funds. You may want to help facilitate those follow-up activities e.g. moving the funds to a shared multi-key wallet or doing a [coinjoin]({{ '/guide/glossary/#coinjoin' | relative_url }}).

## Address compatibility

> There are currently [three different versions]({{ '/guide/glossary/#address' | relative_url }}) of bitcoin addresses. From newest to oldest they are; Segwit, Script, and Legacy. It is the best practice for applications to support the newest address version, SegWit, as default.

Reviewer 4: Add why it is the best practice

Author: we've probably got to add this to the holistic guide review so ill leave as is now, as its used in other places and make a note on #202

> Since "legacy" addresses are still in use[^1], and some older applications have yet to upgrade. Script addresses can be used to resolve incompatibility issues that can arise when the sender's application does not recognize the SegWit address provided by the receiver.

Reviewer 4: Give an example here of a script address, P2WSH maybe.

> To illustrate the problem, suppose the sender is using a legacy wallet and paying to the receiver's SegWit address. In this case, the sender's wallet may incorrectly warn them that the address is invalid or not supported. This can confuse the sender’s end, leading them to think that the receiver provided an incorrect address.

Reviewer 4: ```suggestion
To illustrate the problem, suppose the sender is using a legacy wallet and paying to the receiver's segwit address.
```

Author: @Bosch-0 whats the reason for this simplification?

Reviewer 4: Mostly just following the optech style guide - it is just a personal preference really. We use optech for a lot of references when covering technical details so thought it would be best to go with their styling.

> The receiver should then have the ability to switch to a Script address that does not have all the benefits of SegWit, like cheaper transactions but will be compatible with the sender's wallet.

Reviewer 4: ```suggestion
The receiver should have the ability generate a script SegWit address that does not have all the benefits of native SegWit, like cheaper transactions, but will be compatible with the sender's legacy wallet.
```

Author: So this may not be correct, but I was trying to be consistent with the glossary and other parts of the guide which refer to SegWit (3...) as Script addresses.

Reviewer 4: I feel that leans to the more technical side whereas I see the guide being more for the lay man / designers who may not be so technical

---
layout: guide
title: Receiving bitcoin
description: An introduction to how receiving bitcoin works.
nav_order: 3
parent: Payments
permalink: /guide/payments/receive/
main_classes: -no-top-padding
# image: /assets/images/guide/payments/receive/header.svg
---

# Receiving bitcoin

<!--

Editor's notes

This page should cover what to do when receiving bitcoin, how to share and copy addresses etc.

- What addresses to support (native segwit by default)
- How to present QR codes (allow for specifying an amount)
- Generating a new address 

-->

> For someone to receive bitcoin, they must first provide the sender with a payment destination. They can do this by generating an address within their Bitcoin wallet application. A new address should be generated for every payment, as addresses should only be used once. This helps safeguards the receiver’s privacy.

Reviewer 3: ```suggestion
For someone to receive bitcoin, they must provide the sender with a payment destination. They can do this by generating an address in their Bitcoin wallet application. To help safeguard the receiver’s privacy, a new address should be generated for every payment.
```

<!-- 
Update glossary

Each address that a Bitcoin application generates has an accompaniying private key. This private key allows the receiver to unlock the bitcoin that is sent to the address when the receiver chooses to spend it.

Although addresses are not a problem to share publically, their accompanying private keys which typically generated from a single recovery phrase must be kept secure and private.
-->

> ## Preparing the payment request

Reviewer 3: ```suggestion
## Requesting bitcoin
```

Author: i think having the full term payment* request in there is more useful than not.

> The simplest way to receive bitcoin is by generating and sharing just a Bitcoin address with the sender. Receivers can share the address by either showing it as a scannable QR Code or as plain text that can be shared via any regular communication tool like email, instant messaging, SMS, etc.

Reviewer 3: ```suggestion
The simplest way to receive bitcoin is by generating and sharing a Bitcoin address with the sender. Receivers can share the address by either as a scannable QR code, or as plain text via any regular communication tool like email, instant messaging, SMS, etc.
```

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/payments/receive/simple-share.png"
   retina = "/assets/images/guide/payments/receive/simple-share@2x.png"
   alt-text = "Screen showing how receiver is able to share a bitcoin address as soon as they enter the request flow"
   width = 250
   height = 541
   layout = "float-right-desktop"
%}

</div>

> There are multiple types of addresses. Sharing one that is incompatible with the sender's wallet application can prevent a successful payment. Read more about addresses such [address compatibility]({{ '/guide/glossary/#address-compatibility' | relative_url }}) issues in the glossary.

Reviewer 3: ```suggestion
There are several types of addresses. Sharing one that is incompatible with the sender's wallet application can prevent a successful payment. Read more about [address compatibility]({{ '/guide/glossary/#address-compatibility' | relative_url }}) issues in the glossary.
```

Author: i thought this component after the galley looked a bit messy. what do you think?

<img width="1440" alt="Screenshot 2021-05-12 at 15 23 55" src="https://user-images.githubusercontent.com/183140/117982249-1cef9000-b336-11eb-9603-b88348592a99.png">

Reviewer 3: The text really is a tip though (or warning), so I'd prefer to make it read well.
The next paragraph is also a completely different subject. (Could have it's own sub-heading 'Additional payment details')

Author: nice ye ill add the heading "Inputting additional payment details" similar to how its phrased on the send page

---

> Many Bitcoin wallet applications focus on providing minimalized interfaces for requesting payments. This is convenient in the short-term as it can facilitate fast and easy payments but may not be in the user’s best interest longer-term. Lists of transactions without information about who they are from and their purpose make it hard to manage personal finances and future transaction privacy.

Reviewer 3: ```suggestion
Many Bitcoin wallet applications provide only this most basic interface with an address for requesting payments. This is convenient in the short-term as it is fast and simple, but may not be in the user’s best interest longer-term. Lists of transactions without information about who they are from and their purpose make it hard to manage personal finances and future transaction privacy.
```

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/payments/receive/transactions-no-label-context.png"
   retina = "/assets/images/guide/payments/receive/transactions-no-label-context@2x.png"
   alt-text = "Screen showing a list of transactions with no context just amounts and transaction IDs"
   caption = "When no additional information is gathered from the receiver, only what is stored on the blockchain can be shown."
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/payments/receive/transactions-with-label-context.png"
   retina = "/assets/images/guide/payments/receive/transactions-with-label-context@2x.png"
   alt-text = "Screen showing a list of transactions with details about the payment"
   caption = "While more information can be helpful it requires manual input. Ask which are priorities, and how can you make their entry easier?"
   width = 250
   height = 541
%}

</div>

> Since the Bitcoin blockchain only stores a limited amount of payment information, such as the amount and addresses involved in a transaction, any additional information must be stored in your application. This additional information can be vital in providing context to wallet owners on their spending habits and payment histories. It is possible for receivers to share some of these extra details with senders by encoding them in [payment links]({{ '/guide/foundations/wallet-interoperability/' | relative_url }}). Either way, these details have to be manually inputted when generating a payment request.

Reviewer 3: ```suggestion
Since the Bitcoin blockchain only stores a limited amount of information, such as the amount and addresses involved, any additional information must be stored in your application. This additional information can be vital in providing context to wallet owners on their spending habits and payment histories. These details have to be manually added. It is possible for receivers to share some of these extra details with senders by encoding them in [payment links]({{ '/guide/foundations/wallet-interoperability/' | relative_url }}).
```

The following are the standard properties of a payment link:

- **Address** – The destination of the payment
- **Amount** – How much is being requested (optional)
- **Label** – For example, the name of who is sending the payment (optional)
- **Message** – For example, the purpose of payment (optional)

> Not all applications support storing or even sharing payment links with these details. When providing this functionality, find the right balance between convenience and usefulness for your target audience. Try to also think of ways that you can make inputting these details take less time, so it isn't burdensome.

Reviewer 3: ```suggestion
Not all applications support storing additional information, or sharing payment links with these details. When providing this functionality, balance convenience and usefulness for your target audience. Consider ways that can make the input of these details easy, not burdensome.
```

Author: > Not all applications support storing additional information, or sharing payment links with these details. When providing this functionality, find the right balance between convenience and usefulness for your target audience. Consider ways that can make the input of these details easy, not burdensome.

   retina = "/assets/images/guide/payments/receive/enter-details-purpose@2x.png"
   alt-text = "Screen asking the receiver to enter the purpose of the payment"
   caption = "Sometimes the purpose can be determined by identifying who the payment is from. Be mindful of that, and consider easy ways to help the receiver add more context."
   width = 250
   height = 541
%}

</div>

<!--
Technical nitpick;
Addresses are not exactly "generated"
since they are deterministic technically
what happens is an unused address just gets chosen
-->

<!-- 
Update interopability page
Since there are many Bitcoin applications for people to choose from, its likely that the sender and receiver are not using the same one and they don't have the abilitiy to exchange payment details within the application itself. 

The exchange of payment details then most commonly happens outside of the wallet application. This 

Since bitcoin is a open system and has many payment applications built ontop of it, there is a [standard format]({{ '/guide/foundations/wallet-interoperability/#payment-links' | relative_url }}) of "payment links" that most bitcoin applications use to share payment details.
-->

> ## Sharing the payment request

Reviewer 3: ```suggestion
## Sharing the request
```

Author: also with this one, i think its just clear to have the payment part in here. the payment request concept is something that will be in other places in this chapter I expect so might as well readers familiar with it.

Reviewer 3: You decide. I think it's pretty clear from the page context that it's about payments but up to you.

> Whether it’s just an address, or a payment link containing additional payment information, the two primary methods of sharing payment details are via scannable QR Codes or as plain text.

Reviewer 3: ```suggestion
Whether it’s just an address, or a payment link containing additional information, the two primary methods of sharing requests are via QR Codes or as plain text.
```

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/payments/receive/share.png"
   retina = "/assets/images/guide/payments/receive/share@2x.png"
   alt-text = "Screen showing the invoice with a share button"
   caption = ""
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/payments/receive/share-sharesheet.png"
   retina = "/assets/images/guide/payments/receive/share-sharesheet@2x.png"
   alt-text = "Screen showing the invoice overlayed with a share sheet"
   caption = ""
   width = 250
   height = 541
%}

</div>

**Text**

> Sharing payment details as text is as straightforward as copying and pasting. It is also typical for Bitcoin applications to provide the receiver with a “share” button - allowing them to conveniently share addresses or payment links via other applications, with “share sheets” being offered by most mobile operating systems.

Reviewer 3: ```suggestion
Sharing a request as text is as straightforward as copying and pasting. It is also typical for Bitcoin applications to provide the receiver with a “share” button - allowing them to conveniently share addresses or payment links via other applications, with “share sheets” being offered by most mobile operating systems.
```

**QR Codes**

> QR Codes are also a common way to exchange payment details when the two devices are near one another. It is not without its faults though, as some people may not be familiar with QR Codes, and lower-end phones may have difficulty reading them.

Reviewer 3: ```suggestion
QR Codes are commonly used when the sender and receiver are near one another. It is not without its faults though, as some people may not be familiar with QR Codes, and lower-end phones may have difficulty reading them.
```

Here are some constraints to consider when displaying QR Codes:

- Complex QR Codes can be difficult for devices with low-resolution cameras to read.
- Use high contrast between the QR Code blocks and their background.
- Display the QR Code at the largest size so that the scanner can detect it easily.

_For more technical details about optimizing QR Codes, [see this article](https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses)._

## Awaiting the payment

> Once the receiver's payment details have been shared, senders still have to create and broadcast their transaction, leaving room for uncertainty as to when the payment will be completed and received. Think about how to keep the receiver informed whilst they await their payment.

Reviewer 3: ```suggestion
Once the request has been shared, the sender still has to create and broadcast their transaction, leaving room for uncertainty as to when the payment will be completed and received. Think about how to keep the receiver informed whilst they await their payment.
```

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/payments/receive/receive-with-pending.png"
   retina = "/assets/images/guide/payments/receive/receive-with-pending@2x.png"
   alt-text = "Screen showing a list of incoming payments"
   caption = "By labeling addresses and storing them, you will be able to indicate the state of incoming payments."
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/payments/receive/receive-pending-details.png"
   retina = "/assets/images/guide/payments/receive/receive-pending-details@2x.png"
   alt-text = "Screen showing details of an incoming payment that is pending"
   caption = "Consider some users who are new to Bitcoin may not know how its transactions get processed."
   width = 250
   height = 541
%}

</div>

> If your application has a section where the user can see a list of their payment requests or transaction history, this would be a good place to provide the user with updates on the payment’s status and what stage of completion it is at.

Reviewer 3: ```suggestion
If your application has a list of their payment requests or transaction history, show the status and what stage of completion the transaction is at.
```

> Then once the payment has been finalized, you should also consider what the receiver may want to do with those funds. You may want to help facilitate those follow-up activities for example, moving the funds to a shared multi-key wallet or doing a [coinjoin]({{ '/guide/glossary/#coinjoin' | relative_url }}).

Reviewer 3: ```suggestion
Once the payment has been finalized, consider what the receiver may want to do with those funds. You may want to help facilitate those follow-up activities, for example moving the funds to a shared multi-key wallet or doing a [coinjoin]({{ '/guide/glossary/#coinjoin' | relative_url }}).
```

Reviewer 3: The part about follow-up actions seem very hard to predict. But I'm not opposed to keeping it in here for now.

---
layout: guide
title: Receiving bitcoin
description: An introduction to how receiving bitcoin works.
nav_order: 3
parent: Payments
permalink: /guide/payments/receive/
main_classes: -no-top-padding
# image: /assets/images/guide/payments/receive/header.svg
---

# Receiving bitcoin

<!--

Editor's notes

This page should cover what to do when receiving bitcoin, how to share and copy addresses etc.

- What addresses to support (native segwit by default)
- How to present QR codes (allow for specifying an amount)
- Generating a new address 

-->

> For someone to receive bitcoin, the sender needs to know the recipient’s address. The receiver generates an address in their Bitcoin application and shares it with the sender as text, by scanning a [QR Code]({{ '/guide/foundations/wallet-interoperability/#qr-codes' | relative_url }}). [Payment links]({{ '/guide/foundations/wallet-interoperability/#payment-links' | relative_url }}) are also used to share additional details about a payment and can be shared in the same way.

Reviewer 2: ```suggestion
For someone to receive bitcoin, the sender needs to know the recipient’s address. The receiver generates an address in their Bitcoin application and shares it with the sender. Text, [QR Codes]({{ '/guide/foundations/wallet-interoperability/#qr-codes' | relative_url }}), and [payment links]({{ '/guide/foundations/wallet-interoperability/#payment-links' | relative_url }}) are the most common ways to share those details.
```
There was a bit of duplication in the last two sentences, and `shares it with the sender as text, by scanning a QR Code` also doesn't read well.

Reviewer 4: ```suggestion
For someone to receive bitcoin, the sender needs to know the recipient’s Bitcoin address. The receiver generates an address in their Bitcoin wallet application and shares it with the sender as text, by scanning a [QR Code]({{ '/guide/foundations/wallet-interoperability/#qr-codes' | relative_url }}). [Payment links]({{ '/guide/foundations/wallet-interoperability/#payment-links' | relative_url }}) are also used to share additional details about a payment and can be shared in the same way.
```

> A new address should be generated for every payment, as they should only be used once. This helps safeguards the receiver's privacy.

Reviewer 2: Haven't thought through this, does it also help the senders privacy?

Reviewer 4: I would say yes. It makes the overall tracing of UTXOs more complex

> As the blockchain only stores the amount and addresses involved in a transaction, any additional information has to be stored in your application, or encoded in payment links. It's up to you to decide how and if you allow receivers to enter these additional details. 

Author: somewhat adapted this in the latest updates.

> It's up to you to find the balance between a payment flow that is optimized to quickly generate and share addresses, and one that prompts them to describe the payment so they can use it in the future to have more context about what payment was for.

Reviewer 2: How about something like this:

Many wallet applications focus on a minimal interface for requesting payments. This is convenient in the short-term, but may not be the users best interest longer-term. Lists of transactions without information about who they are from and what the purpose was make it hard to manage personal finances. Try to find the right balance in convenience and usefulness for your target audience.
